.mergify.yml

```diff
@@ -32,6 +33,18 @@ pull_request_rules:
             git merge upstream/{{base}}
             git push upstream {{head}}
             ```
+  - name: close automated pull requests with bump updates if any conflict
+    conditions:
+      - -merged
+      - -closed
+      - conflict
```

Comment:
This seems to assume that the conflict is always caused by another automated bump PR. Are we sure that's always going to be the case?

Reply:
Good point, it's not necessarily the main reason, someone else could actually create a conflict. But I don't think that's normally the case.

Regardless, this particular new rule will filter only automated pull requests with the below conditions:

```
      - author=apmmachine
      - label=automation
```

The idea is to filter those that are coming from humans ( :) ) versus the ones coming from the automated bots.

Although, there is also another author in charge of the bumps when a new release, so that author won't be affected by this rule.

OTOH, I just looked at the history for one of the files and it seems it's normally the automated bot who normally commit changes:

https://github.com/elastic/beats/commits/2fa2616671364bd7e6a5f8c9fc029070a7a92ae4/testing/environments/snapshot-oss.yml

Did I answer your question? Or do you see something else to be considered?

Reply:
You've answered the question. I do think that this creates an edge case where a bump PR might get closed because it conflicts with a non-bump PR but eventual consistency is fine. If we mistakenly close a bump PR another one will come along and take its place or we'll end up having to catch it manually. I think that's probably fine.

Reply:
You've answered the question. I do think that this creates an edge case where a bump PR might get closed because it conflicts with a non-bump PR but eventual consistency is fine. If we mistakenly close a bump PR another one will come along and take its place or we'll end up having to catch it manually. I think that's probably fine.

Reply:
> If we mistakenly close a bump PR another one will come along and take its place or we'll end up having to catch it manually. I think that's probably fine.

Yes, exactly, the automation runs on a daily basis (once a day) and will bump the version only if a new version has been fetched from `artifacts-api.elastic.co`, so the edge case could eventually happen and if so, then next day bump should put things back to normal.

For the record -> https://apm-ci.elastic.co/job/apm-shared/job/bump-stack-version-pipeline/ is the one in charge

I think even if it happens it might be better to clean those automated PRs automatically rather than manually.

Reply:
Cool. I'm on board with this plan. Thanks for the discussion!